python/datashaper/datashaper/workflow.py

```diff
@@ -304,55 +294,49 @@ def output(self, id: Optional[str] = None) -> Table:
     def run(
         self,
         context: Optional[Context] = None,
-        status_reporter: Optional[StatusReporter] = NoopStatusReporter(),
-        create_verb_progress_reporter: Optional[
-            Callable[[str], StatusReportHandler]
-        ] = _create_default_verb_status_reporter,
+        options: WorkflowOptions = WorkflowOptions(),

```

Comment:
rule of thumb is setting objects to None, and create them, if needed and set to None, in the method's body